### PR TITLE
ASoC: SOF: Intel: prepare i915 deferred probe

### PR DESCRIPTION
**Pull request by plbossart**
Lots of small conflicts with existing patches that need to be reverted or updated.
- timeout
- rename of hdev->hda
- move of remove to return void

FYI @kv2019i @marc-hb on the gpu-bind open

### CODE REVIEW

**Comment by plbossart** [comment]
MTLP_RVP_HDA-ace1_0-ipc4 and MTLP_RVP_SDW-ace1_0-ipc4 fail the boot. This is not unexpected given the i915/Xe changes. @kv2019i @marc-hb FYI

**Comment by plbossart** [comment]
https://sof-ci.01.org/linuxpr/PR4645/build460/devicetest/index.html shows MTL failures (despite showing a "green failure" that will make @marc-hb cringe).

**Comment by plbossart** [comment]
with the gpu_bind work-around, results look better on MTL-ACE: 
https://sof-ci.01.org/linuxpr/PR4645/build472/devicetest/index.html

@kv2019i @mlankhorst @marc-hb do the suggested "NOT FOR UPSTREAM" patches meet basic technical sanity?

**Comment by kv2019i** [comment]
@plbossart Ack, this looks great. Still a good question whether it's less work to maintain this here or in the DUT config scripts. One bonus with having this as a not-for-upstream patch here in sof-dev is that every time we do an upstream merge, we can try dropping the workaround and see if i915/xe is mature enough already (the rc1 updates are really the place where we get big i915/xe updates to sof-dev). If yes, drop the workaround, if not, keep it.

PS lnl needs the same thing

**Comment by plbossart** [comment]
I'll go ahead and merge this, and add the LNL part later.

File: sound/soc/sof/intel/mtl.c [hunk 1]
```diff
@@ -672,13 +673,22 @@ static int mtl_dsp_core_put(struct snd_sof_dev *sdev, int core)
 struct snd_sof_dsp_ops sof_mtl_ops;
 EXPORT_SYMBOL_NS(sof_mtl_ops, SND_SOC_SOF_INTEL_HDA_COMMON);
 
+static int mtl_hda_dsp_probe_early(struct snd_sof_dev *sdev)
+{
+	snd_hdac_i915_bind(sof_to_bus(sdev), 0);
+	return hda_dsp_probe_early(sdev);
```
Comment (aiChaoSONG) [inline sound/soc/sof/intel/mtl.c:679]: `sof_to_bus()` referred to `sdev->pdata->hw_pdata`,  but `hw_pdata` is allocated and assigned in `hda_dsp_probe_early()`, is this safe?

Comment (plbossart) [reply sound/soc/sof/intel/mtl.c:679]: You gave me a good scare @aiChaoSONG, but it's fine really: the sof_to_bus(dev) argument is not used.

````c
void snd_hdac_i915_bind(struct hdac_bus *bus, int i915_bind)
{
	gpu_bind = i915_bind;
}
````

so we could remove this argument to make sure it's never used. Care to send a patch?

**Pull request merged**